src/libos/src/signal/do_sigreturn.rs

```diff
@@ -393,47 +379,47 @@ impl Stack {
     }
 }
 
-thread_local! {
-    static PRE_UCONTEXTS: RefCell<CpuContextStack> = Default::default();
+async_rt::task_local! {
+    static SIGNAL_CONTEXT: RefCell<SignalContext> = RefCell::new(SignalContext::new());
 }
 
-#[derive(Debug, Default)]
-struct CpuContextStack {
-    stack: [Option<*mut ucontext_t>; 32],
-    count: usize,
+struct SignalContext {
+    context: *mut ucontext_t,
 }
 
-impl CpuContextStack {
+impl SignalContext {
     pub fn new() -> Self {
-        Default::default()
-    }
-
-    pub fn full(&self) -> bool {
-        self.count == self.stack.len()
+        SignalContext {
+            context: ptr::null_mut(),
+        }
     }
 
-    pub fn empty(&self) -> bool {
-        self.count == 0
-    }
+    pub fn push(&mut self, cpu_context: *mut ucontext_t) {
+        if cpu_context.is_null() {
+            panic!("the signal context is null");
+        }
 
-    pub fn push(&mut self, cpu_context: *mut ucontext_t) -> Result<()> {
-        if self.full() {
-            return_errno!(ENOMEM, "cpu context stack is full");
+        unsafe {
+            (*cpu_context).uc_link = self.context;
```

Comment:
This is cool. I wondered how unlimited nested signals are handled in Linux. But I found `ucountext_t` is poorly documented. I had no idea what is the exact purpose of `uc_link` (I though `getcontext` and `setcontext` may use it) and whether it is safe to us to use it. Could you add a reference link to explain why it is ok to use `uc_link` here?